src/Tests/Microsoft.DotNet.GenAPI.Tests/CSharpFileBuilderTests.cs

```diff
@@ -0,0 +1,520 @@
+                        Blue = 200
+                    }
+                }
+                """);
+        }
+
+        [Fact]
+        public void TestPropertyGeneration()
+        {
+            RunTest(original: """
+                namespace Foo
+                {
+                    public class Car
+                    {
+                        public int? Wheels { get; }
+                        public bool IsRunning { get; set; }
+                    }
+                }
+                """,
+                expected: """
+                namespace Foo
+                {
+                    public partial class Car
+                    {
+                        public int? Wheels { get { throw null; } }
+                        public bool IsRunning { get { throw null; } set { } }
+                    }
+                }
+                """);
+        }
+
+        [Fact]
+        public void TestAbstractPropertyGeneration()
+        {
+            RunTest(original: """
+                namespace Foo
+                {
+                    abstract class Car
+                    {
+                        abstract protected int? Wheels { get; }
+                        abstract public bool IsRunning { get; set; }
+                    }
+                }
+                """,
+                expected: """
+                namespace Foo
+                {
+                    internal abstract partial class Car
+                    {
+                        protected abstract int? Wheels { get; }
+                        public abstract bool IsRunning { get; set; }
+                    }
+                }
+                """);
+        }
+
+        [Fact]
+        public void TestExplicitInterfaceImplementation()
+        {
+            RunTest(original: """
+                namespace Foo
+                {
+                    public interface IControl
+                    {
+                        void Paint();
+                    }
+                    public interface ISurface
+                    {
+                        void Paint();
+                    }
+                
+                    public class SampleClass : IControl, ISurface
+                    {
+                        void IControl.Paint()
+                        {
+                        }
+                        void ISurface.Paint()
+                        {
+                        }
+                    }
+                }
+                """,
+                expected: """
+                namespace Foo
+                {
+                    public partial interface IControl
+                    {
+                        void Paint();
+                    }
+                    public partial interface ISurface
+                    {
+                        void Paint();
+                    }
+                
+                    public partial class SampleClass : IControl, ISurface
+                    {
+                    /*
```

Comment:
Do you need this comment here? It seems unrelated to being an interface implementation test.

Reply:
Thanks, good catch. I updated the UT to
```C#
public interface IControl
{
    void Paint();
}
public interface ISurface
{
    void Paint();
}

public class SampleClass : IControl, ISurface
{
    public void Paint()
    {
    }
}
```